Server/routing/index.js

```diff
@@ -27,39 +27,45 @@ router.get('/', async (ctx) => {
 
 router.get('/t/:miniURL', async (ctx) => {
     // find url in DB or error- does not exist
+    let id = parseInt(ctx.params.miniURL, 36);
+    console.log(id);
+    try {
+        var query = await database.query(
+            'SELECT * FROM public.urls WHERE id = ' + id
+        )
+
+        
+        if (query.rows[0] != null) {
+            await database.query(
+                'UPDATE public.urls SET counter = counter + 1 WHERE id = ' + id
+            )
+            ctx.redirect(query.rows[0].original_url);
+        } else {
+            ctx.body = 'Error: Not a valid url';
+        }
+    } catch (error) {
+        ctx.body = 'Error';
+    }
+
+
     
-    let query = database.query(
-        'SELECT * FROM urls WHERE mini_url = ctx.params.miniURL'
-    )
-    ctx.body = query;
 });
 
 router.post('/minify', async (ctx) => {
     // enter new URL into DB
 
-    ctx.body = ctx.request.body.url;
-
-      /*  Database tables for URLs and users
-
-CREATE TABLE urls(
-   id BIGSERIAL PRIMARY KEY,
-   original_url VARCHAR NOT NULL,
-   mini_url VARCHAR NOT NULL,
-   counter INT NOT NULL,
-   last_hit TIMESTAMP
-   user_id INT
-);
-
+    try {
+        await database.query(
+            `
+            INSERT INTO public.urls(
+            original_url, counter, user_id)
+            VALUES ('` + ctx.request.body.url + `', 0, 0)
```

Comment:
Parameterize queries rather than interpolating values, especially when using values from the outside world. See: https://node-postgres.com/features/queries ("Parameterized query" section).